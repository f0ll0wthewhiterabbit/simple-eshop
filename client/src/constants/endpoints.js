export const PRODUCTS_API = '/data/products.json'
export const USERS_API = '/data/users.json'
