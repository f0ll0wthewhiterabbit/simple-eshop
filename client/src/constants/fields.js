export const STORAGE_FIELD_ACCESS_TOKEN = 'accessToken'
export const STORAGE_FIELD_REFRESH_TOKEN = 'refreshToken'
export const STORAGE_FIELD_THEME = 'theme'
export const FIELD_THEME_DEFAULT = 'default'
export const FIELD_THEME_DARK = 'dark'
export const STORE_FIELD_PRODUCTS = 'products'
export const STORE_FIELD_USERS = 'users'
export const STORE_FIELD_CURRENT_USER = 'usersCurrent'
export const URL_FIELD_RATINGS_FILTER = 'myRatings'
export const URL_FIELD_NO_FILTER = 'all'
