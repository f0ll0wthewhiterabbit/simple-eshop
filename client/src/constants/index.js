export * from './pagePaths'
export * from './endpoints'
export * from './fields'
export * from './roles'
export * from './pageLimits'
