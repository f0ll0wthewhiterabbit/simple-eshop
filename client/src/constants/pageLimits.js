export const DEFAULT_CATALOG_PER_PAGE_LIMIT = 9
export const DEFAULT_ADMIN_PER_PAGE_LIMIT = 10
export const ADMIN_LOW_PER_PAGE_LIMIT = 5
export const ADMIN_HIGH_PER_PAGE_LIMIT = 25
