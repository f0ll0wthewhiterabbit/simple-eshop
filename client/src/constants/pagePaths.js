export const PAGE_PATHS = {
  HOME: '/',
  CATALOG: '/catalog',
  ABOUT: '/about',
  BLOG: '/blog',
  CONTACT: '/contact',
  PROFILE: '/profile',
  SIGN_IN: '/sign-in',
  SIGN_UP: '/sign-up',
  ADMIN: '/admin',
  ADMIN_USERS: '/admin/users',
  ADMIN_PRODUCTS: '/admin/products',
  ADMIN_PRODUCT_ADD: '/admin/products/add',
  ERROR: '/error',
}
