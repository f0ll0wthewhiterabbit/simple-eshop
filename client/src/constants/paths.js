export const HOME_PAGE_PATH = '/'
export const CATALOG_PAGE_PATH = '/catalog'
export const ABOUT_PAGE_PATH = '/about'
export const BLOG_PAGE_PATH = '/blog'
export const CONTACT_PAGE_PATH = '/contact'
export const PROFILE_PAGE_PATH = '/profile'
export const SIGN_IN_PAGE_PATH = '/sign-in'
export const SIGN_UP_PAGE_PATH = '/sign-up'
export const ADMIN_PAGE_PATH = '/admin'
export const ADMIN_USERS_PAGE_PATH = '/admin/users'
export const ADMIN_PRODUCTS_PAGE_PATH = '/admin/products'
export const ADMIN_PRODUCT_ADD_PAGE_PATH = '/admin/products/add'
export const ERROR_PAGE_PATH = '/error'
