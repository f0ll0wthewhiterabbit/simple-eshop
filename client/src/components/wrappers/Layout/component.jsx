import React from 'react'
import PropTypes from 'prop-types'

import AdminLayout from '../../layouts/Admin'
import StandardLayout from '../../layouts/Standard'
import { ROLE_ADMIN } from '../../../constants'

const Layout = ({ isAuthenticated, userRole, children }) => {
  if (isAuthenticated && userRole === ROLE_ADMIN) {
    return <AdminLayout>{children}</AdminLayout>
  }

  return <StandardLayout>{children}</StandardLayout>
}

Layout.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  userRole: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
}

export default Layout
