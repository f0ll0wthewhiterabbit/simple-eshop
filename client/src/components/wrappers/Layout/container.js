import { connect } from 'react-redux'

import Layout from './component'

const mapStateToProps = state => ({
  isAuthenticated: state.getIn(['auth', 'isAuthenticated']),
  userRole: state.getIn(['auth', 'user', 'role']),
})

export default connect(mapStateToProps)(Layout)
