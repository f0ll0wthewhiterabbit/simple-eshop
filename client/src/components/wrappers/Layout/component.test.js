import React from 'react'
import { shallow } from 'enzyme'

import Layout from './component'
import { ROLES } from '../../../constants'
import findByTestAttr from '../../../utils/findByTestAttr'

describe('Layout component', () => {
  let wrapper
  const testContent = 'test content'
  const initialProps = {
    isAuthenticated: false,
    userRole: ROLES.GUEST,
    children: <p>{testContent}</p>,
  }
  const dataTestStandardLayout = 'standardLayout'
  const dataTestAdminLayout = 'adminLayout'

  const generateWrapper = passedProps => {
    const defaultProps = { ...initialProps }
    const props = { ...defaultProps, ...passedProps }

    return shallow(<Layout {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render standardLayout', () => {
    const standardLayout = findByTestAttr(wrapper, dataTestStandardLayout)

    expect(standardLayout).toHaveLength(1)
  })

  it('should contain correct children', () => {
    const standardLayout = findByTestAttr(wrapper, dataTestStandardLayout)

    expect(standardLayout.find('p').text()).toEqual(testContent)
  })

  it('should render adminLayout for administrator', () => {
    wrapper = generateWrapper({ isAuthenticated: true, userRole: ROLES.ADMIN })
    const adminLayout = findByTestAttr(wrapper, dataTestAdminLayout)

    expect(adminLayout).toHaveLength(1)
  })

  it(`shouldn't render adminLayout for authenticated user`, () => {
    wrapper = generateWrapper({ isAuthenticated: true, userRole: ROLES.USER })
    const adminLayout = findByTestAttr(wrapper, dataTestAdminLayout)

    expect(adminLayout).toHaveLength(0)
  })
})
