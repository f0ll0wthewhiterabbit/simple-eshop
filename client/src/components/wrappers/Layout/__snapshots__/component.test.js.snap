// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render correctly 1`] = `
<Connect(StandardLayout)
  data-test="standardLayout"
>
  <p>
    test content
  </p>
</Connect(StandardLayout)>
`;
